skbio/metadata/intersection.pyx

```diff
@@ -0,0 +1,513 @@
+"""
```

Comment:
I'm not opposed, but something that @jairideout and I have been talking a lot about recently is creating our own index objects for tabular-ish datastructures (in light of the issues we are experiencing in having our codebase so closely coupled with pandas). If we were to go that route, representing this IntervalTree as sane index-like object could make a lot of sense for the codebase generally. I'm imagining indexes would have meaningful dispatch and provide a dict-like interface (unlike pandas indexes). So it could be used as "just" an IntervalTree, but it would also plug into things like TabularMSA and FeatureTable.

That being said, scikit-bio indexes might not happen for a while (or ever).